packit/distgit.py

```diff
@@ -100,35 +104,45 @@ def commit(self, title: str, msg: str, prefix: str = "[packit] ") -> None:
         main_msg = f"{prefix}{title}"
         self.local_project.git_repo.git.add("-A")
         self.local_project.git_repo.index.write()
+        commit_args = ["-s", "-m", main_msg]
+        if msg:
+            commit_args += ["-m", msg]
         # TODO: attach git note to every commit created
         # TODO: implement cleaning policy: once the PR is closed (merged/refused), remove the branch
         #       make this configurable so that people know this would happen, don't clean by default
         #       we should likely clean only merged PRs by default
         # TODO: implement signing properly: we need to create a cert for the bot, distribute it to the container,
         #       prepare git config and then we can start signing
         # TODO: make -s configurable
-        self.local_project.git_repo.git.commit("-s", "-m", main_msg, "-m", msg)
+        self.local_project.git_repo.git.commit(*commit_args)
 
-    def push_to_fork(self, branch_name: str, fork_remote_name: str = "fork"):
+    def push_to_fork(self, branch_name: str, fork_remote_name: str = "fork", force: bool = False):
         """
         push changes to a fork of the dist-git repo; they need to be committed!
 
         :param branch_name: the branch where we push
         :param fork_remote_name: local name of the remote where we push to
+        :param force: push forcefully?
         """
         if fork_remote_name not in [
             remote.name for remote in self.local_project.git_repo.remotes
         ]:
-            fork_urls = self.local_project.git_project.get_fork().get_git_urls()
+            fork = self.local_project.git_project.get_fork()
+            if not fork:
+                self.local_project.git_project.fork_create()
+                fork = self.local_project.git_project.get_fork()
```

Comment:
I am not sure if we can be sure, about the absence of the delay. The ogr call to API should wait for the finish, but if we find some problems, we can fix that on the OGR site lately.

Reply:
Huh? You mean there is a race condition when the API call is done and the fork is still not ready?

Reply:
I am just not sure... It's more a note for myself to test it properly in OGR.